tests/round_combinations/debug_path_differences.py

```
@@ -0,0 +1,94 @@
+
+        # Check 2-cycles
+        for neighbor in neighbors:
+            if neighbor in TRANSACTION_GRAPH and node in TRANSACTION_GRAPH[neighbor]:
+                print(f"  2-cycle: {node} <-> {neighbor}")
+
+    # Print some example paths to see patterns
+    print("\nExample paths of each length:")
+    for length in sorted(dfs_by_length.keys()):
+        length_paths = [p for p in paths if len(p) == length]
+        print(f"\nLength {length} (showing first 3):")
+        for i, path in enumerate(length_paths[:3]):
```

Ruff (0.11.9)

Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)